### PR TITLE
SHS-4047: A11y: Adjust slide transition based on user's Reduced Motion Preference

### PR DESCRIPTION
**Pull request by mariannuar**
# READY FOR REVIEW

## Summary
- Adjust slide transition based on user's Reduced Motion Preference in the following components:
  - Banner Image(s) with Partial Overlay (with multiple slides)
  - Banner Image(s) with Text Box (with multiple slides)
  - Spotlight (with multiple slides)
  - Photo Album (in Slider mode) 

## Need Review By (Date)
02/09

## Urgency
medium

## Steps to Test
1. For each component, you need to go to your `System Settings` > `Accessibility` > `Display`
2. Turn on and off the Reduce motion in all the pages below

### Banner Image(s) with Partial Overlay (with multiple slides)
- [ ] For traditional, go to https://hs-traditional.suhumsci.loc/
- [ ] For colorful, go to https://hs-colorful.suhumsci.loc/componentshero-images-main-region/hero-gradient-slider
- [ ] Do step 1 and 2

### Banner Image(s) with Text Box (with multiple slides)
- [ ] For traditional, go to https://hs-traditional.suhumsci.loc/hero-images-main-region/hero-layered-slider
- [ ] For colorful, go to https://hs-colorful.suhumsci.loc/hero-layered-slider
- [ ] Do step 1 and 2

### Spotlight (with multiple slides)
- [ ] For traditional, go to https://hs-traditional.suhumsci.loc/componentshero-images-main-region/tall-spotlight 
- [ ] Edit it and add more spotlights in the component
- [ ] For colorful, go to https://hs-colorful.suhumsci.loc/componentshero-images-main-region/tall-spotlight
- [ ] Do the same
- [ ] Do step 1 and 2

### Photo Album (in Slider mode) 
- [ ] For traditional, go to https://hs-traditional.suhumsci.loc/components/photo-album
- [ ] For colorful, go to https://hs-colorful.suhumsci.loc/components/photo-album
- [ ] Do step 1 and 2

## PR Checklist
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)
- [Humsci Basic PR Checklist](https://github.com/SU-HSDO/suhumsci/blob/develop/docs/HumsciBasicPRChecklist.md)

### CODE REVIEW

File: docroot/themes/humsci/humsci_basic/src/js/shared/prefered-reduced-motion/prefered-reduced-motion.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+        // Check if the user prefers reduced motion
+        if (!mediaQuery || mediaQuery.matches) {
+          $(slickInstance).on('beforeChange', (event, slick) => {
+            slick.slickSetOption('cssEase', 'none', true);
+          });
+        } else {
+          $(slickInstance).on('beforeChange', (event, slick) => {
+            slick.slickSetOption('cssEase', 'ease', true);
+          });
```
Comment (mariannuar) [inline docroot/themes/humsci/humsci_basic/src/js/shared/prefered-reduced-motion/prefered-reduced-motion.js]: While testing more, I found out the library already does this by default (remove animations at all). Test in a different branch/env.

Comment (mariannuar) [reply docroot/themes/humsci/humsci_basic/src/js/shared/prefered-reduced-motion/prefered-reduced-motion.js]: Here's a video https://www.loom.com/share/857ec27d89124493a8f6e14dc3bfe4c4?sid=622fc92f-3bf5-4020-9968-2ea143965b6a

Comment (mariannuar) [reply docroot/themes/humsci/humsci_basic/src/js/shared/prefered-reduced-motion/prefered-reduced-motion.js]: Here's how it looks with the fade transition https://www.loom.com/share/60fb89ed8d80476da5dec32de826b5f8?sid=16e771f5-96cb-4796-a1c0-da7c8edfd068

File: docroot/themes/humsci/humsci_basic/src/js/shared/prefered-reduced-motion/prefered-reduced-motion.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
+(function ($, Drupal, window) {
+  Drupal.behaviors.mySlickBehavior = {
+    // eslint-disable-next-line no-unused-vars
+    attach(context, settings) {
```
Comment (cienvaras) [inline docroot/themes/humsci/humsci_basic/src/js/shared/prefered-reduced-motion/prefered-reduced-motion.js]: If you're not using the `settings` argument, just remove it. That way you don't need the `eslint-disable`

**Review by cienvaras — CHANGES_REQUESTED** [review]
@mariannuar Works great! Just fix a minor coding issue and we're done 🙂

**Review by cienvaras — APPROVED** [review]
🚀

**Pull request merged**